File: backend/src/main/java/com/equisoft/dca/backend/project/model/BackupKeepInterval.java
package com.equisoft.dca.backend.project.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum BackupKeepInterval implements BaseEnum<Integer> {
	DAYS(1, "Days"),
	WEEKS(2, "Weeks"),
	MONTHS(3, "Months");

	private final int id;

	private final String name;

	BackupKeepInterval(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/BackupKeepInterval.ts
import BaseEnum from './BaseEnum';

export default class BackupKeepInterval extends BaseEnum {

    public static DAYS = new BackupKeepInterval('1', 'project:backupsEnum:days');
    public static WEEKS = new BackupKeepInterval('2', 'project:backupsEnum:weeks');
    public static MONTHS = new BackupKeepInterval('3', 'project:backupsEnum:months');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}


